code/modules/clothing/gloves/color.dm

```diff
@@ -12,6 +12,45 @@
 	custom_price = 1200
 	custom_premium_price = 1200
 
+/obj/item/toy/sprayoncan
+	name = "spray-on insuls applicator"
+	desc = "What is the number one problem facing our station today?"
+	icon = 'icons/obj/clothing/gloves.dmi'
+	icon_state = "sprayoncan"
+
+/obj/item/toy/sprayoncan/afterattack(atom/target, mob/living/carbon/user, proximity)
+	if(iscarbon(target) && proximity)
+		var/mob/living/carbon/C = target
+		var/mob/living/carbon/U = user
+		if(C == user)
+			C.visible_message("<span class='notice'>[U] sprays their hands with glittery rubber!</span>")
+		else
+			C.visible_message("<span class='warning'>[U] sprays glittery rubber on the hands of [C]!</span>")
+		C.equip_to_slot_or_del(new /obj/item/clothing/gloves/color/yellow/sprayon,ITEM_SLOT_GLOVES) //we're gonna pretend that the rubber can't stick to gloves.
+		qdel(src)
+
+/obj/item/clothing/gloves/color/yellow/sprayon
+	desc = "How're you gonna get 'em off, nerd?"
+	name = "spray-on insulated gloves"
+	icon_state = "sprayon"
+	item_state = "sprayon"
+	permeability_coefficient = 0
+	resistance_flags = ACID_PROOF
+	var/shocks_remaining = 10
+
+/obj/item/clothing/gloves/color/yellow/sprayon/Initialize()
+	.=..()
+	ADD_TRAIT(src, TRAIT_NODROP, CURSED_ITEM_TRAIT) //Cursed with the power of science, or something
+
+/obj/item/clothing/gloves/color/yellow/sprayon/proc/Shocked()
+	shocks_remaining--
+	if(shocks_remaining < 0)
+		qdel(src) //if we run out of uses, the gloves crumble away into nothing, just like my dreams after working with .dm
+
+/obj/item/clothing/gloves/color/yellow/sprayon/dropped()
+	.=..()
+	qdel(src) //loose nodrop items bad

```

Comment:
There is a DROP_DEL flag in item_flags. Use that instead.

Reply:
dropped() is better than dropdel for this because dropdel only deletes if it hits the floor, instead of being moved from its slot at all